### ISSUE TITLE
Mounting/dismounting rockets in the Space Station dimension

### ISSUE THREAD

**Issue by EzerArch** [micdoodle8/Galacticraft#1592]
I'm getting some inconsistencies with mounting and dismounting rockets in the Space Station dimension.

Player is airborne and right-click rocket:
- Tier 1: Player enters and immediately kicked out.
- Tier 2 and 3: Player enters and stays in.

Player is on solid block and right-clicks rocket:
- All tiers: Player enters and stays in.

Player is in the rocket, presses LSHIFT to dismount (game always prints ```Press LSHIFT to dismount```):
- Tier 1: Worked
- Tier 2 and 3: Not worked, player doesn't sneak either.

Player is in the rocket, right-clicks to dismount (*):
- All tiers: Worked

(*) In my opinion, for consistency, this should be disabled since this is as well in the other vehicles (buggy, horses and minecarts etc), no matter the dimension.

**Comment by EzerArch** [comment]
Related to #1892?

**Comment by radfast** [comment]
Here's what I'm seeing currently in 423:

**Mounting rockets:**
* works in the same way in all dimensions: Right Click mounts
* works in the same way with all tiers of rocket
* the hitbox of the rockets (the clickable part) is smaller than you expect, you can also notice this if breaking the rocket - especially, the Tier 1 rocket has a small hitbox - this means that it needs some accuracy to mount a Tier 1 rocket if you are freefalling / flying

**Dismounting rockets:**
On Overworld / Moon / Mars:
* Left Shift (sneak key) and Right Click both dismount the rocket
* Left Shift leaves you in 3rd person view after you dismount, Right Click switches you to 1st person view - I actually like this behaviour
* You are initially positioned very high when you dismount - near the nose of the rocket - with the Tier 3 rocket it is so large you can take half a heart of fall damage

On Space Stations:
* Same except that the Left Shift key is disabled (this is connected with motion on Space Stations, also my attempts to totally disable sneaking on Space Stations)

I will look at fixing the behaviour on Space Stations.  Maybe also the hitboxes if I can find a safe way to do it, the problem is that the hitboxes are also connected with landing https://github.com/micdoodle8/Galacticraft/commit/dc14036affa686021688c6ea9329c1785d63bace.  The fall damage I think I'm OK with for now, rockets are a little bit dangerous to play around with.

**Comment by radfast** [comment]
I'm going to leave the hitbox issue alone for now.  It's on my 'eventually to do' list, but it seems minor and nobody is really complaining about it.

**Comment by dj3520** [comment]
Unfortunately I've noticed this in my survival server when I figured my space station was a good platform to prepare my tier 2 rocket. I wanted to make sure it was fully fueled so I got in and could not get back out. Luckily enough of my plan was already done that I wasn't completely screwed, yet it was everything from graceful after landing. Then there was the problem of how I'd get back and forth between my space station since I had disabled cross-dimensional teleports and relied on Sync to get around that (won't transfer items.) But now my space station had no clone there! Wasn't a huge deal since I had prepared a backup plan, but having to go through with it because I couldn't escape a rocket?
